src/systems/variational_smoother_constraint.C

```diff
@@ -521,6 +521,47 @@ void VariationalSmootherConstraint::constrain_node_to_line(const Node & node,
     }
 }
 
+void VariationalSmootherConstraint::find_nodal_or_face_neighbors(
+    const MeshBase &mesh, const Node &node,
+    const std::unordered_map<dof_id_type, std::vector<const Elem *>>
+        &nodes_to_elem_map,
+    std::vector<const Node *> &neighbors)
+{
+  // Find all the nodal neighbors... that is the nodes directly connected
+  // to this node through one edge.
+  MeshTools::find_nodal_neighbors(mesh, node, nodes_to_elem_map, neighbors);
+
+  // If no neighbors are found, use all nodes on the containing side as
+  // neighbors.
+  if (!neighbors.size())
+    {
+      // Grab the element containing node
+      const auto *elem = nodes_to_elem_map.at(node.id()).front();
+      // Find the element side containing node
+      for (const auto &side : elem->side_index_range())
+        {
+          bool found_node = false;
+          const auto &nodes_on_side = elem->nodes_on_side(side);
+          for (const auto &local_node_id : nodes_on_side)
+            if (elem->node_id(local_node_id) == node.id())
+            {
+              found_node = true;
+              break;
+            }
```

Comment:
I'd refactor this into a `std::find_if(nodes_on_side.begin(), nodes_on_side.end(), [&](auto local_node_id){...});` call rather than hand-coding the linear search.